### ISSUE TITLE
after fired turbo:render x-data not exists

### ISSUE THREAD

**Issue by zodman** [SimoTod/alpine-turbo-drive-adapter#36]
Hi guys, thanks for the connector, but a found a little issue:

I had this form:
![image](https://user-images.githubusercontent.com/44167/116957532-ac71b080-ac5d-11eb-99bf-e8aad22a06a6.png)
as you can see the alpinejs works at the beginning. when the form submitted is succeed turbo reload the complete page.

![image](https://user-images.githubusercontent.com/44167/116957321-3d945780-ac5d-11eb-8a0d-a995dfd966c0.png)
and the alpinejs not working ....

What happened ?

The turbo:render fired but not found any dom node related alpine  `document.querySelectorAll('[x-data]')`

If I put on console the snippet

```
      window.Alpine.discoverUninitializedComponents((el) => {
        window.Alpine.initializeComponent(el)
      })
```
Now the alpinejs works...

TLDR; when turbo:render the ``document.querySelectorAll('[x-data]')` return empty  and the window.Alpine.initializeComponent never happened.

![image](http://g.recordit.co/RSHoST09Nh.gif)

**Comment by SimoTod** [comment]
@zodman Could you let me know if it was working with 1.0.4, please?
I recently switched to turbo:render from turbo:load to address some issues when forms returns a 423 response and I just want to make sure it's not a regression.
If you can also post a minimal example to reproduce (it's time consuming and error prone to copy the html from a gif), it would help a lot

**Comment by zodman** [comment]
yes `  "alpine-turbo-drive-adapter": "^1.0.4",`

**Comment by SimoTod** [comment]
@zodman Mmh, so it was working correctly with 1.0.4. Interesting.
I just need the HTML of the page (possibly only the relevant bits) in a comment here, otherwise it's a bit of a nightmare to get it from the gif you posted.

**Comment by SimoTod** [comment]
So... the issue is because you use a turbo-frame with an src attribute which is loaded after turbo triggered `turbo:render` and even `turbo:load`. 1.0.4 was briefly working as a side effect of a different fix because I had put an additional refresh 2 frames after any form was submitted but that, for instance, wouldn't work with normal links triggering a turbo-frame update.
Unfortunately, it doesn't seem like there is an event to capture when a turbo frame loads. Looking at the turbo repo, there are people asking for it but the maintainers don't seem to be keen on introducing new events because they are ways to do it with Stimulus and they are not too interested in supporting different frameworks
I'll keep digging into it but you might need to manually trigger the refresh for now.

**Comment by zodman** [comment]
@SimoTod  do you have the link of the way to do it with stimulus ?

**Comment by SimoTod** [comment]
I don't have a link, unfortunately. I saw someone suggesting it on a turbo bug. I'm going to work on a solution but it will likely to be end of next week.

**Comment by zodman** [comment]
@SimoTod what about a timeout function for fired before the turbo:render finish

**Comment by SimoTod** [comment]
I'd rather avoid time based behaviors because they feel hacky, they are hard to test and they are even harder to debug when you have an issue. For example, a lazy loaded turbo frame is an http request so, on slow connection, it might take longer then the arbitrary timeout we pick. I'll implement something based on mutation observers.

**Comment by SimoTod** [comment]
Hi @zodman could you test https://cdn.jsdelivr.net/gh/simotod/alpine-turbo-drive-adapter@feature/turbo-frame-support/dist/alpine-turbo-drive-adapter.min.js and let me know if it fixes your issue? Thanks

**Comment by lobre** [comment]
Is this still an issue for you @zodman? The PR seems still on hold waiting for your feedback. Thanks.

**Comment by SimoTod** [comment]
This seems to work how of the box with Alpine 3, can anyone confirm? (cc @lobre)

**Comment by lobre** [comment]
I was just asking news before choosing turbo + alpine for a project (to be sure it was mature enough), so I was not exactly suffering from this problem.

Maybe @zodman can confirm? Otherwise, as you said it seems OK to me with the new version at least.
